### PR TITLE
Added support for methods from map.h and tileset.h for the Python plugin API.

### PR DESCRIPTION
**Pull request by yashgandhe666**
Having problems with generating pythonbind.cpp

### CODE REVIEW

**Comment by bjorn** [comment]
> Having problems with generating pythonbind.cpp

What problems are those exactly?

On Arch Linux, I have installed `python2-pybindgen` and use `python2 tiledbinding.py` to regenerate the `pythonbind.cpp` file.

**Comment by harindu95** [comment]
I have the same error. With PyBindGen (0.18.0).
> pybindgen.typehandlers.base.CodeGenerationError: Class QRgb cannot be copied

Traceback:
>python2 tiledbinding.py
Traceback (most recent call last):
  File "tiledbinding.py", line 312, in <module>
    mod.generate(fh)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/module.py", line 928, in generate
    self.do_generate(sink_manager, module_file_base_name)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/module.py", line 702, in do_generate
    submodule.do_generate(out)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/module.py", line 767, in do_generate
    class_.generate(sink, self)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/cppclass.py", line 2003, in generate
    self._generate_methods(code_sink, parent_caller_methods)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/cppclass.py", line 2394, in _generate_methods
    utils.call_with_error_handling(overload.generate, (code_sink,), {}, overload)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/utils.py", line 180, in call_with_error_handling
    return callback(*args, **kwargs)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/overloading.py", line 154, in generate
    (code_sink,), {}, self.all_wrappers[0])
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/utils.py", line 180, in call_with_error_handling
    return callback(*args, **kwargs)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/cppmethod.py", line 387, in generate
    self.generate_body(tmp_sink, gen_call_params=[class_])
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/typehandlers/base.py", line 936, in generate_body
    self.return_value.convert_c_to_python(self)
  File "/home/harindu/VirtualENVs/py2/local/lib/python2.7/site-packages/pybindgen/cppclass.py", line 3254, in convert_c_to_python
    raise CodeGenerationError("Class {0} cannot be copied".format(self.cpp_class.full_name))
pybindgen.typehandlers.base.CodeGenerationError: Class QRgb cannot be copied

**Comment by bjorn** [comment]
Hmm, I get this error as well with PyBindGen 0.18.0. With 0.17 it worked fine, so you can either look for the problem or downgrade to PyBindGen 0.17. I'll try to look into this soon.

**Comment by bjorn** [comment]
@yashgandhe666 @harindu95 Issues should be fixed in change da9ba8f0c7138850b7c2b9b23e6544d0705a94ae.

File: src/plugins/python/tiledbinding.py [hunk 1]
```diff
@@ -77,7 +77,11 @@
 cls_tileset.add_method('tileHeight', 'int', [])
 cls_tileset.add_method('tileSpacing', 'int', [])
 cls_tileset.add_method('margin', 'int', [])
-#cls_tileset.add_method('tileOffset', 'QPoint', [])
+cls_tileset.add_method('tileOffset', 'QPoint', [])
+cls_tileset.add_method('gridSize', 'Qsize', [])
+cls_tileset.add_method('setGridSize', None, ('Qsize', 'gridSize'))
```
Comment (bjorn) [inline src/plugins/python/tiledbinding.py]: Should be `QSize` (though currently this class is not defined, nor is `QPoint`, which is probably why the `tileOffset` method is commented out).

**Comment by bjorn** [comment]
@yashgandhe666 In my last update I've used PyBindGen 0.18 to update the generated file. Please make sure you're using 0.18 as well, to avoid the huge differences in the generated file between 0.17 and 0.18.

**Pull request closed without merging**